libpkg/pkg_solve.c

```diff
@@ -1136,6 +1156,9 @@ pkg_solve_sat_problem(struct pkg_solve_problem *problem)
 				if (var->flags & PKG_VAR_FAILED) {
 					var->flags ^= PKG_VAR_INSTALL | PKG_VAR_FAILED;
 				}
+				if (var->unit->pkg->vital) {
```

Comment:
Please add a comment here to explicit why you are doing this, in a decade it won't be so obvious ;)